prov/tcp/src/tcpx_progress.c

```diff
@@ -341,22 +349,23 @@ static int tcpx_get_rx_entry(struct tcpx_rx_detect *rx_detect,
 		rx_entry->msg_hdr = rx_detect->hdr;
 		rx_entry->msg_hdr.hdr.op_data = TCPX_OP_READ_RSP;
 		rx_entry->done_len = sizeof(rx_detect->hdr);
+		new_rx->process_fn = process_rx_read_entry;
 		break;
 	default:
 		FI_WARN(&tcpx_prov, FI_LOG_DOMAIN,
 			"invalid op in hdr\n");
 		return -FI_EINVAL;
 	}
```

Comment:
You might be able to create an array of function pointers, assign those to the ep, and use the opcode as an index into the array (after validating it).  That would avoid the switch statement completely.